src/Qowaiv.Validation.DataAnnotations/NestedValidationContext.cs

```diff
@@ -53,11 +53,11 @@ internal class NestedValidationContext : IServiceProvider
     private readonly List<IValidationMessage> collection;
 
     /// <summary>Adds a set of messages.</summary>
-    public void AddMessages(IEnumerable<ValidationResult> messages)
+    public void AddMessages(IEnumerable<ValidationResult> messages, bool onType = false)
```

Comment:
Renamed it to `violationOnType`. Is handles path to the property. If the violation is on the type (instead of the property) the member name should be constructed differentially.